python/spider-py/src/spider_py/task_executor/task_executor.py

```diff
@@ -1,8 +1,171 @@
+        if param.annotation is inspect.Parameter.empty:
+            msg = f"Parameter {param.name} has no type annotation."
+            raise TypeError(msg)
+        if cls is client.Data:
+            if not isinstance(arg, bytes):
+                msg = f"Argument {i} for spider.Data is not bytes."
+                raise TypeError(msg)
+            core_data = store.get_data(UUID(bytes=arg))
+            parsed_args.append(client.Data(core_data))
+        else:
+            parsed_args.append(from_serializable(cls, arg))
+    return parsed_args
+
+
+def parse_results(results: object) -> list[object]:
+    """
+    Parses results from the function execution.
+    :param results: Results to parse.
+    :return: Parsed results.
+    :raises TypeError: If a result cannot be parsed.
+    """
+    response_messages: list[object] = [TaskExecutorResponseType.Result]
+    if isinstance(results, tuple):
+        for result in results:
+            if isinstance(result, client.Data):
+                response_messages.append(result.id.bytes)
+            else:
+                response_messages.append(to_serializable(result))
+    elif isinstance(results, client.Data):
+        response_messages.append(results.id.bytes)
+    else:
+        response_messages.append(to_serializable(results))
+    return response_messages
+
 
 def main() -> None:
-    """Main function to execute the task."""
+    """
+    Main function to execute the task.
+    :raises StorageError: If a storage operation fails.
+    """
+    # Parses arguments
+    args = parse_args()
+    func = args.func
+    task_id = args.task_id
+    task_id = UUID(task_id)
+    libs = args.libs
+    storage_url = args.storage_url
+    input_pipe_fd = args.input_pipe
+    output_pipe_fd = args.output_pipe
+
+    logger.debug("Function to run: %s", func)
+
+    # Sets up storage
+    storage_params = storage.parse_jdbc_url(storage_url)
+    store = storage.MariaDBStorage(storage_params)
+
+    with fdopen(input_pipe_fd, "rb") as input_pipe, fdopen(output_pipe_fd, "wb") as output_pipe:
+        input_message = receive_message(input_pipe)
+        arguments = get_request_body(input_message)
+        logger.debug("Args buffer parsed")
+
+        # Get the function to run
+        function_name = func.split(".")[-1] if "." in func else func
+        function = None
+        for lib in libs:
+            module = importlib.import_module(lib)
+            if hasattr(module, function_name):
+                function = getattr(module, function_name)
+                break
+        if function is None:
+            msg = f"Function {function_name} not found in provided libraries."
+            raise ValueError(msg)
+
+        signature = inspect.signature(function)
+        if len(signature.parameters) != len(arguments) + 1:
+            msg = (
+                f"Function {function_name} expects {len(signature.parameters) - 1} "
+                f"arguments, but {len(arguments)} were provided."
```

Comment:
```suggestion
                f"Function {function_name} expects {len(signature.parameters) - 1} arguments, but"
                f" {len(arguments)} were provided."
```
- We wrap when the line limit reaches 100.
- We usually add space on the second line.